src/DataProvider/ResourceProvider.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * Copyright (c) 2017-2020 Andreas Möller
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE.md file that was distributed with this source code.
+ *
+ * @see https://github.com/ergebnis/test-util
+ */
+
+namespace Ergebnis\Test\Util\DataProvider;
+
+use Ergebnis\Test\Util;
+
+final class ResourceProvider
+{
+    use Util\Helper;
+
+    /**
+     * @return \Generator<string, array{0: resource}>
+     */
+    public static function resource(): \Generator
+    {
+        yield from self::provideDataForValues([
+            'resource' => \fopen(__FILE__, 'rb'),
```

Comment:
This might result in fatal errors when too many files are open, but we can fix it when it does.